SW/Boiling Detection/boil_detect.py

```diff
@@ -170,71 +164,76 @@ def update(self, frame, stove):
         :param frame: Input frame used to update boiling status
         :param stove: Stove object holding current stove state
         """
-        # updates frame_count
-        self.frame_count = self.frame_count + 1
-        burners = self.__burners(stove)
+        # time function begins
+        start_time = time.time()
+
+        # updates frames_elapsed
+        self.frames_elapsed = self.frames_elapsed + 1
+        burners = stove.get_burners()
 
         # creates new masks if a pot was added
         self.initialize_masks(burners)
 
         # if program has not executed before
-        if self.first_image is None:
-            self.first_image = frame
-            self.first_quads = self.__split_frame(self.first_image)
+        if self.last_image is None:
+            self.last_image = frame
+            self.last_quads = self.__split_frame(self.last_image)
 
         # if the number of frames that have been processed are desired amount, proceed
-        elif self.frame_count == self.count:
+        elif self.frames_elapsed == self.frame_frequency:
             # if there is no second image recorded yet
-            if self.second_image is None:
-                self.second_image = frame
-                self.second_quads = self.__split_frame(self.second_image)
+            if self.current_image is None:
+                self.current_image = frame
+                self.current_quads = self.__split_frame(self.current_image)
             # second image does exist, must shift images
             else:
-                self.first_image = self.second_image
-                self.second_image = frame
+                self.last_image = self.current_image
+                self.current_image = frame
 
             # splits the frames
-            self.first_quads = self.__split_frame(self.first_image)
-            self.second_quads = self.__split_frame(self.second_image)
+            self.last_quads = self.__split_frame(self.last_image)
+            self.current_quads = self.__split_frame(self.current_image)
 
             # Detects boiling on active burners
-            i = 0
+            curr_idx = 0 # index corresponding to current burner
             for burner in burners:
```

Comment:
```
for burner, burner_ssim, burner_boiling in zip(burners, median_SSIM, boiling):
	# rest of code -- avoids indexing
```

Reply:
okay wait quick question, for some of these though I need to modify the element stored at a certain index. So, wouldn't I still need an index variable in some cases bc this is creating a local variable? I did include in the zip the variables I wasn't modifying

Reply:
are you actually changing the reference, or are you just mutating the object?  if the latter, you can mutate the object without an index. (like push/pop elements onto the array)

Reply:
in the places where i use an index, I believe I want to be changing the reference not just mutating the object